src/client.ts

```diff
@@ -369,6 +446,36 @@ export class Client {
   // FILE MARKER - METHODS - PRIVATE ///////////////////////////////////////////
   //////////////////////////////////////////////////////////////////////////////
 
+  /**
+   * This method is used internally to calculate the element Viewport (Dimensions)
+   * @param selector
+   * @returns ViewPort object
+   */
+  private async getViewport(selector: string) {
+    let values: DOMRect;
+    try {
+      values = JSON.parse(
+        await this.evaluatePage(
+          `JSON.stringify(document.querySelector('${selector}').getBoundingClientRect())`,
+        ) as string,
+      );
+    } catch (error) {
+      await this.done();
+      throw new Error(
+        "Selector supplied doesn't return any element, JSON Parse error" +
+          error.name,
+      );
+    }
+    const viewPort: ViewPort = {
+      x: values.x,
+      y: values.y,
+      width: values.width,
+      height: values.height,
+      scale: 2,
+    };
```

Comment:
I think this whole function could be written in a slightly different way, to avoid a try catch and to properly handle an error
```ts
const result = await evaluatePage(...) // Pass in an invalid selector so you know what the error response is
if ("success" in result === false && "error" in res) { // test case needed to cover this
  await this.done()
  this.checkForErrorResult(result)
}
const values: DOMRect = JSON.parse(result)
return { // no need to declare a variable just to immediantly return it
  x: ...,
  y: ...
}  

```

Reply:
need some discussion on this

Reply:
the closest I could get to an exception "and/or" a valid value was this
```typescript
const res = await this.sendWebSocketMessage("Runtime.evaluate",{
      expression: `JSON.stringify(document.querySelector('${selector}').getBoundingClientRect())`
    }) as {
      result: {
        type: "string";
        value?: string;
      };
    } | { // Present if we get a `cannot read property 'value' of null`, eg if `selector` is `input[name="fff']`
      result: Exception
    };
```

there is no exceptionDetail, and if I just try for getting DOMRect, value is not coming in the returned object
which means i cannot use checkForErrorResult